### PR TITLE
fix: webui remove checkpoint button for deleted checkpoint [DET-4286]

### PR DESCRIPTION
**Pull request by apizzini**
## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

**Review by hamidzr — COMMENTED** [review]
this is looking good. Do we know if we also want the filter to take this change into account? I think we probably do.

before
Screen Capture_select-area_20201020104038

after

Screen Capture_select-area_20201020104056

**Comment by apizzini** [comment]
@hamidzr your question is if we want to filter out trials that (now) doesn't have the checkpoint button when "has checkpoint" filter is selected?

I don't know, maybe @hkang1 can reply?

**Comment by hkang1** [comment]
> @hamidzr your question is if we want to filter out trials that (now) doesn't have the checkpoint button when "has checkpoint" filter is selected?
> 
> I don't know, maybe @hkang1 can reply?

@apizzini  Let's treat the rows with deleted checkpoints the same as rows without checkpoints. 🙏

**Comment by apizzini** [comment]
Added the filter.

File: webui/react/src/pages/TrialDetails.tsx [hunk 1]
```diff
@@ -140,7 +140,7 @@ const TrialDetailsComp: React.FC = () => {
 
   const columns = useMemo(() => {
     const checkpointRenderer = (_: string, record: Step) => {
-      if (record.checkpoint) {
+      if (record.checkpoint && record.checkpoint.state !== CheckpointState.Deleted) {
```
Comment (hamidzr) [inline webui/react/src/pages/TrialDetails.tsx:143]: nit: suggestion: capture this repeated bit a `hasCheckpoint` helper

**Review by hamidzr — APPROVED** [review]

File: webui/react/src/pages/TrialDetails.tsx (hunk 1 above)
Comment (apizzini) [reply webui/react/src/pages/TrialDetails.tsx:143]: Done, I'll merge when you lmk is good.

Comment (hamidzr) [reply webui/react/src/pages/TrialDetails.tsx]: thanks for updating this :rocket: :ship:

**Pull request merged**